types/src/transaction.rs

```diff
@@ -0,0 +1,785 @@
+mod auction_transaction;
+mod direct_call;
+mod error;
+mod native_transaction;
+mod pricing_mode;
+#[cfg(any(all(feature = "std", feature = "testing"), test))]
+mod test_transaction_builder;
+mod transaction_approval;
+#[cfg(any(feature = "std", test))]
+mod transaction_builder;
+mod transaction_hash;
+mod transaction_header;
+mod transaction_kind;
+mod userland_transaction;
+
+use alloc::{collections::BTreeSet, vec::Vec};
+use core::{
+    cmp,
+    fmt::{self, Debug, Display, Formatter},
+    hash,
+};
+
+#[cfg(feature = "datasize")]
+use datasize::DataSize;
+#[cfg(feature = "json-schema")]
+use once_cell::sync::Lazy;
+#[cfg(any(feature = "once_cell", test))]
+use once_cell::sync::OnceCell;
+#[cfg(feature = "json-schema")]
+use schemars::JsonSchema;
+#[cfg(any(feature = "std", test))]
+use serde::{Deserialize, Serialize};
+use tracing::debug;
+
+#[cfg(any(all(feature = "std", feature = "testing"), test))]
+use crate::testing::TestRng;
+use crate::{
+    bytesrepr::{self, FromBytes, ToBytes},
+    crypto, Digest, DisplayIter, PublicKey, SecretKey, TimeDiff, Timestamp,
+};
+#[cfg(any(feature = "std", test))]
+use crate::{AccountAndSecretKey, TransactionConfig};
+pub use auction_transaction::AuctionTransaction;
+pub use direct_call::DirectCall;
+pub use error::{
+    DecodeFromJsonError as TransactionDecodeFromJsonError, Error as TransactionError,
+    ExcessiveSizeError as TransactionExcessiveSizeError, TransactionConfigFailure,
+};
+pub use native_transaction::NativeTransaction;
+pub use pricing_mode::PricingMode;
+#[cfg(any(all(feature = "std", feature = "testing"), test))]
+pub use test_transaction_builder::TestTransactionBuilder;
+pub use transaction_approval::TransactionApproval;
+#[cfg(any(feature = "std", test))]
+pub use transaction_builder::{TransactionBuilder, TransactionBuilderError};
+pub use transaction_hash::TransactionHash;
+pub use transaction_header::TransactionHeader;
+pub use transaction_kind::TransactionKind;
+pub use userland_transaction::UserlandTransaction;
+
+#[cfg(feature = "json-schema")]
+static TRANSACTION: Lazy<Transaction> = Lazy::new(|| {
+    let secret_key = SecretKey::example();
+    let args = crate::runtime_args! { "amount" => 1000 };
+    Transaction::build(
+        *Timestamp::example(),
+        TimeDiff::from_seconds(3_600),
+        PricingMode::GasPriceMultiplier(1),
+        String::from("casper-example"),
+        TransactionKind::from(NativeTransaction::new_mint_transfer(args)),
+        AccountAndSecretKey::SecretKey(secret_key),
+    )
+});
+
+/// A signed smart contract.
```

Comment:
I feel like this line "A signed smart contract" is some ancient holdover from devnet

Reply:
Done in [633530d](https://github.com/casper-network/casper-node/pull/4130/commits/633530d1a7ad8d8957b9363ae6935a9318d6694c).